index.js

```diff
@@ -1,29 +1,112 @@
 'use strict';
 
-const {Adapter, Device} = require('gateway-addon');
+const {Adapter, Device, Property} = require('gateway-addon');
 const wol = require('wol');
+const findDevices = require('local-devices');
+const {promise: ping} = require('ping');
 
 class WakeOnLanAdapter extends Adapter {
   constructor(addonManager, manifest) {
     super(addonManager, manifest.name, manifest.name);
     addonManager.addAdapter(this);
 
-    for (const mac of manifest.moziot.config.devices) {
-      this.handleDeviceAdded(new WakeOnLanDevice(this, mac));
+    this.checkPing = manifest.moziot.config.checkPing || false;
+
+    findDevices().then((devices) => {
+      for (const mac of manifest.moziot.config.devices) {
+        const arpDevice = devices.find((d) => d.mac === mac.toLowerCase());
```

Comment:
If the ARP table isn't fully populated, this may end up returning `undefined`, in which case, you'll try to add an undefined device. You should probably stub an object to pass to `addDevice()` in that case.